### ISSUE TITLE
Support selectRange on PluralRules

### ISSUE THREAD

**Issue by hsivonen** [unicode-org/icu4x#3012]
Add support for https://github.com/tc39/proposal-intl-numberformat-v3/issues/16 to ICU4X.

**Comment by jedel1043** [comment]
I was trying to implement `PluralRules` for Boa and I saw this API wasn't implemented.
I'll try to contribute this :)

**Comment by sffc** [comment]
Cool! A few tips to get you started:

1. We need to start consuming `pluralRanges.json` from CLDR-JSON:
    1. Add it here: https://github.com/unicode-org/icu4x/blob/687d0fff1a1581d8473eb1246655be0671ebd365/tools/testdata-scripts/globs.rs.data#L32
    2. Then run `cargo make download-repo-sources`
2. Now you can add a data struct for the plural ranges into `icu_plurals::provider`. This is the most interesting step because you need to design a data structure that is correct, compact, and efficient. It is usually _not_ a good idea to directly copy over the JSON structure.
3. Once that is ready, add a transformer for the data in datagen/src/transform/cldr/plurals, then add the data key to plurals/data/config.json + datagen/src/registry.rs, then you can run `cargo make testdata` and `cargo make bakeddata components/plurals` to see it all transform before your eyes
4. Finally, you can build the actual API for the feature. I'd suggest a new struct called `PluralRanges` in the icu_plurals crate.

Each of these steps can be done as a separate PR if you prefer.

This process is all documented in writing_a_data_struct_md.
